### PR TITLE
ci: switch to GitHub Actions

### PR DESCRIPTION
**Pull request by troyronda**
Closes: #307

Signed-off-by: Troy Ronda <troy@troyronda.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -3,29 +3,32 @@
...
 
     - uses: actions/checkout@v1
 
-    - name: Unit Test
-      timeout-minutes: 5
+    - name: Run unit test
       run: make unit-test
+    
+    - name: Upload coverage to Codecov  
+      uses: codecov/codecov-action@v1.0.2
```
Comment (soluchok) [inline .github/workflows/build.yml]: Add `timeout-minutes: 5` per job
and move it to `Run checks` job bc here we have matrix strategy and it (Codecov) doesn't work for mac

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -3,29 +3,32 @@
...
       uses: actions/setup-go@v1
       with:
         go-version: 1.13
       id: go
 
     - uses: actions/checkout@v1
 
-    - name: Unit Test
-      timeout-minutes: 5
+    - name: Run unit test
```
Comment (soluchok) [inline .github/workflows/build.yml]: Add `timeout-minutes: 5` as well

Comment (troyronda) [reply .github/workflows/build.yml]: timeout goes with the job not the step AFAIK.
https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

(this was the change I made).

File: .github/workflows/build.yml (hunk 1 above)
Comment (troyronda) [reply .github/workflows/build.yml]: codecov uses the result of unit test run. I had to duplicate the unit test section.

Comment (troyronda) [reply .github/workflows/build.yml]: (I left the matrix so the OS is filled into the check name).

File: .github/workflows/build.yml (hunk 2 above)
Comment (troyronda) [reply .github/workflows/build.yml]: The documentation is unclear on what the default step timeout is...
https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes

File: .github/workflows/build.yml (hunk 1 above)
Comment (troyronda) [reply .github/workflows/build.yml]: replaced with a conditional.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -3,54 +3,64 @@
...
 jobs:
-
   unitTest:
-    name: Unit Test
+    name: Unit test
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
     strategy:
       matrix:
         os: [ubuntu-latest, macOS-latest]
```
Comment (llorllale) [inline .github/workflows/build.yml]: Suggest we pin the OS versions

File: .github/workflows/build.yml (hunk 2 above)
Comment (troyronda) [reply .github/workflows/build.yml]: We will need to revisit the timeouts. I just copied 10mins into each long step and job.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -3,54 +3,64 @@
...
 
     - uses: actions/checkout@v1
 
-    - name: Unit Test
-      timeout-minutes: 5
+    - name: Run unit test
+      timeout-minutes: 10
       run: make unit-test
 
+    # TODO: need secret for Codecov from Actions
```
Comment (llorllale) [inline .github/workflows/build.yml]: Doesn't look like we have the necessary option under this project's Settings tab: https://help.github.com/en/articles/virtual-environments-for-github-actions#creating-and-using-secrets-encrypted-variables

File: .github/workflows/build.yml (hunk 3 above)
Comment (troyronda) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 4 above)
Comment (troyronda) [reply .github/workflows/build.yml]: Yup - that's why I left CircleCI unit test step enabled.

**Pull request merged**